src/partition.jl

```diff
@@ -0,0 +1,150 @@
+mutable struct Partition
+    num_districts::Int
+    num_cut_edges::Int
+    total_population::Int
+
+    district_populations::Array{Int, 1} # of length(num_districts)
+    cut_edges::Array{Int, 1}            # of length(num_edges)
+    populations::Array{Int, 1}          # of length(num_nodes)
+    assignments::Array{Int, 1}          # of length(num_nodes)
+    district_adj::SparseMatrixCSC{Int, Int}
+    district_nodes::Array{Set{Int}, 1}
+    attributes::Array{Dict{String, Any}}
+end
+
+function Partition(graph::Graph,
+                   raw_graph::Dict{String, Any},
+                   pop_col::AbstractString,
+                   assignment_col::AbstractString,
+                   nodes_str::AbstractString = "nodes")::Partition
+    """ Builds the Partition Struct, which stores all the information of our
```

Comment:
The discussion of secret-keeping here is informative and probably necessary, but we should think about what tone we want our documentation to take—let's take a look at the GerryChain documentation (which is quite good!) together and come up with some loose guidelines?

Reply:
A discussion of data structures is not necessary - user-specific docstrings would be more helpful.